BlueprintUICommonControls/Sources/Rule.swift

```diff
@@ -0,0 +1,77 @@
+import BlueprintUI
+import UIKit
+
+/// A solid line, parallel to the x or y axis, with a fixed thickness but unconstrained in length,
+/// suitable for use as a separator.
+public struct Rule: ProxyElement {
+    /// Represents whether the rule is parallel to the x or y axis.
+    public enum Orientation {
+        /// Indicates that the rule is parallel to the x axis.
+        case horizontal
+        /// Indicates that the rule is parallel to the y axis.
+        case vertical
+    }
+
+    /// Represents the thickness of a rule in the direction perpendicular to its orientation.
+    public enum Thickness {
```

Comment:
Should we conform this to `ExpressibleByIntegerLiteral` and `ExpressibleByFloatLiteral`? Then consumers could write:

```swift
let rule = Rule(orientation: .horizontal, color: .red, thickness: 2)
```

Doing so loses some specificity (i.e. is it 2 points or pixels?), but it might be worth that trade-off.

Reply:
This is an alright convenience, but I don't think it removes any of the magic from `hairline`.

Reply:
Fwiw, I prefer the enum route, so it's easy for us to find where custom sizes are used.

Reply:
I expect that `.hairline` will be the most commonly used option. If we want to improve the ergonomics of a custom line width I would try adding a convenience initializer before adding an `EspressibleByIntegerLiteral` conformance.

1. It makes it really easy to find code that calls that initializer in the future.
2. I'm very hesitant to add `ExpressibleBy[Foo]Literal` conformance to types that are not... literally a `Foo` (an integer/number in this case); it just makes for non-obvious code that someone/you will have to dig into in the future. In practice I've always ended up removing that stuff down the road for the sake of clarity.